hono/test/utils/factory/items.ts

```diff
@@ -1,64 +1,129 @@
-export function validItem() {
-  return {
-    name: 'Vintage Denim Pants',
-    brand: 'Levi',
-    photoUrl: 'https://example.com/',
-    type: 'top',
-    rating: 2,
+import { faker } from '@faker-js/faker'
+import { drizzle } from 'drizzle-orm/postgres-js'
+import postgres from 'postgres'
+import * as schema from '../../../src/schema'
+import { items, itemTypeEnum } from '../../../src/schema'
+
+export type ItemType = 'layer' | 'top' | 'bottom' | 'footwear' | 'accessory'
+
+export interface Item {
+  id: string
+  name: string
+  brand: string
+  photoUrl: string
+  type: ItemType
+  rating: number
+  createdAt: Date
+  authorUsername: string
+}
+
+export interface ItemAPI {
+  id: string
+  name: string
+  brand: string
+  photoUrl: string
+  type: ItemType
+  rating: number
+  createdAt: string
+  authorUsername: string
+}
+
+export class ItemFactory implements Item {
+  constructor(seed?: number, options?: Partial<Item> | ItemAPI) {
+    faker.seed(seed ?? undefined)
+    this.id = options?.id
+      ? (options.id as string)
+      : faker.string.alphanumeric({ length: 24, casing: 'lower' })
+    this.name = options?.name ? (options.name as string) : faker.commerce.productName()
+    this.brand = options?.brand ? (options.brand as string) : faker.company.name()
+    this.photoUrl = options?.photoUrl
+      ? (options.photoUrl as string)
+      : faker.image.urlLoremFlickr({ category: 'fashion' })
+    this.type = options?.type
+      ? (options.type as ItemType)
+      : (faker.helpers.arrayElement(itemTypeEnum) as ItemType)
+    this.rating = options?.rating
+      ? (options.rating as number)
+      : faker.number.int({ min: 0, max: 4 })
+    this.createdAt = options?.createdAt
+      ? new Date(options.createdAt as Date)
+      : new Date(faker.date.past().toISOString().split('T')[0])
+    this.authorUsername = options?.authorUsername
+      ? (options.authorUsername as string)
+      : faker.internet.userName()
+  }
+
+  async store(db_url: string) {
+    const db = drizzle(postgres(db_url), { schema })
```

Comment:
Might be a good idea to abstract this behavior into the db.ts file for consistency.

Reply:
Addressed, closing.